src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs

```diff
@@ -127,18 +127,56 @@ public static bool TryCreateNativeIntegerType(INamedTypeSymbol symbol, out TypeS
 
 #if !CODE_STYLE
 
+#nullable enable
             public override TypeSyntax VisitFunctionPointerType(IFunctionPointerTypeSymbol symbol)
             {
-                // TODO(https://github.com/dotnet/roslyn/issues/39865): generate the calling convention once exposed through the API
+                FunctionPointerCallingConventionSyntax? callingConventionSyntax = null;
+                if (symbol.Signature.CallingConvention != System.Reflection.Metadata.SignatureCallingConvention.Default)
+                {
+                    SeparatedSyntaxList<FunctionPointerUnmanagedCallingConventionSyntax> conventionsList = default;
+                    switch (symbol.Signature.CallingConvention)
```

Comment:
Consider using pattern matching here, it might reduce the length?

Reply:
I'm not sure what you mean, I don't really see a way to make this smaller with pattern matching.

---
In reply to: [463854020](https://github.com/dotnet/roslyn/pull/46467#discussion_r463854020) [](ancestors = 463854020)